src/WidgetBase.ts

```diff
@@ -421,8 +464,8 @@ export class WidgetBase<P extends WidgetProperties = WidgetProperties, C extends
 		return this._cachedVNode;
 	}
 
-	public invalidate(): void {
-		if (this._renderState === WidgetRenderState.IDLE) {
+	public invalidate(force?: boolean): void {
```

Comment:
You are correct @agubler ! Normally the widget won't schedule a re-render if there is already a render happening, but some meta require a DOM node to work correctly, so we need to render twice in order to provide one.